### DISCUSSION TITLE
How are binding affinity predictions performed?

### DISCUSSION THREAD

**Discussion by tony-res** [martinez-zacharya/TRILL#23] [Q&A] [answered]
> Next, you need to download the Smina binary to perform binding affinity prediction

The online docs talk about a tool to perform binding affinity prediction. Is there an example of how to use Trill for this?

Is there any tool you'd recommend for predicting the free energy of an antibody-antigen binding complex given a prediction from ESMFold?

**Comment by martinez-zacharya** [comment, marked as answer]
As of the current version of TRILL (1.3.8), it'll automatically use Smina to score your ranked poses after performing diffdock. I plan on adding a feature where you can just submit a complex to trill and it'll use just smina to score, but it's still in the works. You would essentially need to provide the antibody as the protein and the antigen as the ligand and then use "dock". After DiffDock outputs your ranked poses, it will take each one and score it with smina and append the binding affinity in kcal/mol to the structure file name. For example, "docked_6lu7_rank1_confidence-3.36_189.10283.pdb" will be the docked protein and ligand, with the DiffDock confidence score as 3.36 and the smina binding affinity is the 189.10283.

 If the antigen is a protein, the trill won't be able to perform the docking yet, I still am working on integrating DiffDock-PP.

Regarding tools for predicting free energy, you can use smina by itself like so-
1. First you need to convert your protein and ligand into pdbqt files using Open Babel (downloadable via conda)
  `obabel antibody.pdb -O antibody.pdbqt`
  `obabel antigen.pdb -O antigen.pdbqt`
2. Next you can simply use smina to perform the scoring
  `./smina --score_only -r antibody.pdbqt -l antigen.pdbqt`

This requires that you've found your optimal docking position. If you haven't, you will need to use smina to dock the complex for you, which is possible but I haven't tried before. Smina may not be SOTA for predicting free energy, but I just like how permissive and easy to use it is. Let me know if you have any trouble!

**Reply by tony-res** [reply]
It looks like you have LightDock working for the Docking. I'm having some trouble and I think the fix is pretty easy. It looks like maybe the swarm directories aren't being created. Does that sound right?

![image](https://github.com/martinez-zacharya/TRILL/assets/124098125/a25a2361-db6a-4142-afe0-287ad981f094)

**Comment by tony-res** [comment]
Wonderful. I really appreciate this tool. It brings everything together nicely.
